src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -3108,6 +3110,483 @@ def __call__(self, *_inputs: T_OTXDataEntity) -> T_OTXDataEntity | None:
+        inputs.bbox_scale = inputs.bbox_scale * scale
+        inputs.bbox_rotation = rotate
+
+        return inputs
+
+    def __repr__(self) -> str:
+        """Print the basic information of the transform.
+
+        Returns:
+            str: Formatted string.
+        """
+        repr_str = self.__class__.__name__
+        repr_str += f"(shift_prob={self.shift_prob}, "
+        repr_str += f"shift_factor={self.shift_factor}, "
+        repr_str += f"scale_prob={self.scale_prob}, "
+        repr_str += f"scale_factor={self.scale_factor}, "
+        repr_str += f"rotate_prob={self.rotate_prob}, "
+        repr_str += f"rotate_factor={self.rotate_factor})"
+        return repr_str
+
+
+class TopdownAffine(tvt_v2.Transform, NumpytoTVTensorMixin):
+    """Get the bbox image as the model input by affine transform.
+
+    Required Keys:
+
+        - img
+        - bbox_center
+        - bbox_scale
+        - bbox_rotation (optional)
+        - keypoints (optional)
+
+    Modified Keys:
+
+        - img
+        - bbox_scale
+
+    Added Keys:
+
+        - input_size
+        - transformed_keypoints
+
+    Args:
+        input_size (Tuple[int, int]): The input image size of the model in
+            [w, h]. The bbox region will be cropped and resize to `input_size`
+    """
+
+    def __init__(self, input_size: tuple[int, int], is_numpy_to_tvtensor: bool = False) -> None:
+        super().__init__()
+
+        self.input_size = input_size
+        self.is_numpy_to_tvtensor = is_numpy_to_tvtensor
+
+    @staticmethod
+    def _fix_aspect_ratio(bbox_scale: np.ndarray, aspect_ratio: float) -> np.ndarray:
+        """Reshape the bbox to a fixed aspect ratio.
+
+        Args:
+            bbox_scale (np.ndarray): The bbox scales (w, h) in shape (n, 2)
+            aspect_ratio (float): The ratio of ``w/h``
+
+        Returns:
+            np.darray: The reshaped bbox scales in (n, 2)
+        """
+        w, h = np.hsplit(bbox_scale, [1])
+        return np.where(w > h * aspect_ratio, np.hstack([w, w / aspect_ratio]), np.hstack([h * aspect_ratio, h]))
+
+    @staticmethod
+    def _get_warp_matrix(
+        center: np.ndarray,
+        scale: np.ndarray,
+        rot: float,
+        output_size: tuple[int, int],
+        shift: tuple[float, float] = (0.0, 0.0),
+        inv: bool = False,
+        fix_aspect_ratio: bool = True,
+    ) -> np.ndarray:
+        """Calculate the affine transformation matrix that can warp the bbox area.
+
+        Args:
+            center (np.ndarray[2, ]): Center of the bounding box (x, y).
+            scale (np.ndarray[2, ]): Scale of the bounding box
+                wrt [width, height].
+            rot (float): Rotation angle (degree).
+            output_size (np.ndarray[2, ] | list(2,)): Size of the
+                destination heatmaps.
+            shift (0-100%): Shift translation ratio wrt the width/height.
+                Default (0., 0.).
```

Comment:
I think type annotation should be added and `0-100%` is moved to the description.

Reply:
6185ebc626a001974975b35d9535747253d9361d